### PR TITLE
Add rpc to get next mq seq in txpool.

### PR DESCRIPTION
**Pull request by kvinwang**
Part 3 of #414

### CODE REVIEW

File: standalone/rpc-ext/src/lib.rs [hunk 1]
```diff
@@ -123,134 +75,44 @@ where
         + ProvideRuntimeApi<Block>,
     Client::Api:
         sp_api::Metadata<Block> + ApiExt<Block, StateBackend = backend::StateBackendFor<BE, Block>>,
+	Client::Api: MqApi<Block>,
```
Comment (h4x3rotab) [inline standalone/rpc-ext/src/lib.rs:78]: Whitespace

File: standalone/rpc-ext/src/mq_seq.rs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        None => return Ok(None),
+    };
+
+    log::debug!(target: "rpc-ext", "State seq for {}: {}", sender, seq);
+
+    // Now we need to query the transaction pool
+    // and find transactions originating from the same sender.
+    //
+    // Since extrinsics are opaque to us, we look for them using
+    // `provides` tag. And increment the nonce if we find a transaction
```
Comment (h4x3rotab) [inline standalone/rpc-ext/src/mq_seq.rs:66]:
```suggestion
    // `provides` tag. And increment the sequence if we find a transaction
```

File: standalone/rpc-ext/src/mq_seq.rs [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            current_seq,
+            hex::encode(&current_tag),
+            tx.provides().iter().map(|x| format!("{}", hex::encode(x))).collect::<Vec<_>>(),
+        );
+        // since transactions in `ready()` need to be ordered by sequence
+        // it's fine to continue with current iterator.
+        for tg in tx.provides() {
+            if tg == &current_tag {
+                current_seq += 1;
+                current_tag = tag(&sender, current_seq);
```
Comment (h4x3rotab) [inline standalone/rpc-ext/src/mq_seq.rs:83]: After we found the provided tag we can break this for loop, though seq should be the the second provided tag.

File: standalone/rpc-ext/src/mq_seq.rs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    let api = client.runtime_api();
+    let best_hash = client.info().best_hash;
+    let at = BlockId::hash(best_hash);
+
+    let seq = match api
+        .sender_sequence(&at, &sender)
+        .or(Err(Error::SenderNotFound))?
+    {
+        Some(seq) => seq,
+        None => return Ok(None),
```
Comment (h4x3rotab) [inline standalone/rpc-ext/src/mq_seq.rs:57]: It's possible that sender's OffchainIngress is None. In this case it equal to 0. We should still scan the txpool I believe.

File: standalone/rpc-ext/src/mq_seq.rs [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    let sender_scl = hex::decode(sender_hex).map_err(|_| Error::InvalidSender)?;
+    let sender = MessageOrigin::decode(&mut &sender_scl[..]).map_err(|_| Error::InvalidSender)?;
+
+    let api = client.runtime_api();
+    let best_hash = client.info().best_hash;
+    let at = BlockId::hash(best_hash);
+
+    let seq = match api
+        .sender_sequence(&at, &sender)
+        .or(Err(Error::SenderNotFound))?
```
Comment (h4x3rotab) [inline standalone/rpc-ext/src/mq_seq.rs:54]: In which case this could return an error?

File: standalone/rpc-ext/src/mq_seq.rs (hunk 4 above)
Comment (kvinwang) [reply standalone/rpc-ext/src/mq_seq.rs:57]: indeed, fixed

File: standalone/rpc-ext/src/mq_seq.rs (hunk 3 above)
Comment (kvinwang) [reply standalone/rpc-ext/src/mq_seq.rs:83]: Done

File: standalone/rpc-ext/src/lib.rs (hunk 1 above)
Comment (kvinwang) [reply standalone/rpc-ext/src/lib.rs]: fixed

File: standalone/rpc-ext/src/mq_seq.rs (hunk 2 above)
Comment (kvinwang) [reply standalone/rpc-ext/src/mq_seq.rs]: Fixed

File: standalone/rpc-ext/src/mq_seq.rs (hunk 5 above)
Comment (kvinwang) [reply standalone/rpc-ext/src/mq_seq.rs]: It's some low-level error, such as codec errror, generated by the proc_macro.

Comment (h4x3rotab) [reply standalone/rpc-ext/src/mq_seq.rs]: OK. Good to leave them here

**Review by h4x3rotab — APPROVED** [review]
LGTM

**Pull request merged**